optaplanner-core/src/main/java/org/optaplanner/core/api/score/buildin/bendable/BendableScoreHolder.java

```diff
@@ -218,6 +219,11 @@ public void reward(RuleContext kcontext, int[] hardWeightsMultiplier, int[] soft
      * @param weight higher is better, negative for a penalty, positive for a reward
      */
     public void addHardConstraintMatch(RuleContext kcontext, int hardLevel, int weight) {
+        if (hardLevel >= hardScores.length) {
+            throw new IllegalArgumentException("The hardLevel (" + hardLevel
```

Comment:
Soft suggestion: use a template for this repetitive error message and just pass the parameters that change.

Reply:
That might be an option; this does not block merging, though.

Please see also my other comment regarding unit testing.